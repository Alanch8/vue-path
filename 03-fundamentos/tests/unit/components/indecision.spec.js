import { shallowMount } from '@vue/test-utils'
import Indecision from '@/components/Indecision.vue'

describe('Indecision component', () => {

    let wrapper;
    let clgSpy;
    
    global.fetch = jest.fn()

    beforeEach( () => {
        wrapper = shallowMount( Indecision )

        clgSpy = jest.spyOn( console, 'log' )

        jest.clearAllMocks()

    })
    
    test('debe de hacer match con el snapshot', () => {
        expect( wrapper.html() ).toMatchSnapshot()
    })

    test('escribir en el input no debe de disparar nada (console.log)', async() => {

        const getAnswerSpy = jest.spyOn( wrapper.vm, 'getAnswer' )

        const input = wrapper.find('input')
        await input.setValue('Hello world?')

        expect( clgSpy ).toHaveBeenCalledTimes(1)
        // expect( getAnswerSpy ).toHaveBeenCalledTimes(0)
        expect( getAnswerSpy ).not.toHaveBeenCalled()

    })

    test('escribir el simbolo de "?" debe de disparar el getAnswer', async() => {
        const getAnswerSpy = jest.spyOn( wrapper.vm, 'getAnswer' )

        const input = wrapper.find('input')
        await input.setValue('Hello world?')

        expect( clgSpy ).toHaveBeenCalledTimes(1)
        // expect( getAnswerSpy ).toHaveBeenCalledTimes(0)
        expect( getAnswerSpy ).toHaveBeenCalled()
    })

    test('pruebas en el getAnswer', () => {

    })

    test('pruebas en getAnswer - Fallo en el API', () => {

    })

})