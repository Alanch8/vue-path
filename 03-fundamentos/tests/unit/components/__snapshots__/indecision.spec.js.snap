// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Indecision component debe de hacer match con el snapshot 1`] = `
<!--v-if-->
<div class="bg-dark"></div>
<div class="indecision-container"><input type="text" placeholder="Ask me a question!">
  <p>Remember to end with a question mark (?)</p>
  <!--v-if-->
</div>
`;
